### ISSUE TITLE
GM-PHD in Motion detector

### ISSUE THREAD

**Issue by Nuzhny007** [blefaudeux/gmphd#2]
Hi!
It's not an issue - just a question.
I'm interesting in area of motion detectors. You can see the project Multitarget-tracker: https://github.com/Smorodov/Multitarget-tracker
I'm one of contributors. And it will be interesting to use your project for the comparing quality and performance of the different algorithms of the detection and tracking.
May be you can give me some recommends for the integration GM-PHD to the our project? We have some regions (contours) after Background subtraction. And after...?
In this file I added function GMPHDTracker with GMPHD (default parameters from your test): https://github.com/Nuzhny007/Multitarget-tracker/blob/gmphd/main.cpp
GM-PHD is the replacement of the Hungrian+Kalman? If you can give me some advices it will be cool.
Thank you!

**Comment by blefaudeux** [comment]
Hi  Sergey,
sorry for the delay, super busy right now. GMPHD and Hungrian+Kalman do not serve the same purpose if I'm not mistaken, since GMPHD does not handle the association problem. They're complimentary though, because GMPHD handles probabilistic noise (which is not the case for Hungrian, if I'm not mistaken again). In terms of questions that the algorithms answer, I would describe them (+KF), as:
- my detectors give noisy measurements, not just in terms of metrics but also detection rate, which means that I have false positive and misses. GMPHD will try to solve the question : "how many targets do I really have there, and where are they, given those noisy inputs ?" It doesn't say how they associate with a previous measurement though, but it handles targets disappearing momentarily, targets spawning new targets, and false detections 
- I get reasonable detection over time, but don't know how to match them with another. Then my understanding is that it's something that Hungrian will solve
- I get noisy measurements, in terms of metrics, for the same random variable. What's its real trajectory, and foreseeable future ? That's what the Kalman solves 

In a nutshell (I don't have too much time unfortunately), it looks like GMPHD could be useful indeed, as a first step in your pipeline to try to clean up false detections, and the rest of the pipeline can stay the same

**Comment by blefaudeux** [comment]
p.s. the video is pretty neat !

**Comment by Nuzhny007** [comment]
Hi!
Thanks for the full answer! I try to use your GMPHD implementation after blobs detection algorithm. After I will write you about results.
